gadget/update.go

```diff
@@ -1474,6 +1496,7 @@ func canUpdateStructure(fromVss []VolumeStructure, fromIdx int, toVss []VolumeSt
 		if !from.HasFilesystem() {
 			return fmt.Errorf("cannot change a bare structure to filesystem one")
 		}
+		// Note that if partial filesystem this will compare empty strings
```

Comment:
we are not allowing to go from partial no filesystem to filesystem and the other way around? do we need tests for all these combinations?

Reply:
I was thinking on not allowing changes if partial is defined only in one of from/to, but I've changed this now so going from partial to defined is ok - we should allow making the gadget more concrete I think. I've also added unit tests covering the different cases.